### PR TITLE
[UpSample nearest]

### PR DESCRIPTION
**Pull request by jjsjann123**
1. update on restricting block.z <= 64, compliant to CUDA maximum z-dimension of
a block;
2. clang-format

### CODE REVIEW

**Comment by ngimel** [comment]
test case please?

File: aten/src/ATen/native/cuda/UpSampleNearest2d.cu [hunk 1]
```diff
@@ -147,7 +151,8 @@ static void upsample_nearest2d_out_cuda_template(
   // upsample_2d_shape_check makes sure input/output tensor is not empty;
   int block_x = std::min<int>(lastPow2(output_width), max_threads);
   int block_y = std::min<int>(lastPow2(output_height), max_threads / block_x);
-  int block_z = std::min<int>(nc, max_threads / block_x / block_y);
+  int block_z =
+      std::min<int>(64, std::min<int>(nc, max_threads / block_x / block_y));
```
Comment (ngimel) [inline aten/src/ATen/native/cuda/UpSampleNearest2d.cu]: Can you please query maxThreadDims?  
https://docs.nvidia.com/cuda/cuda-runtime-api/structcudaDeviceProp.html#structcudaDeviceProp_192d195493a9d36b2d827aaf3ffd89f1e

**Review by soumith — CHANGES_REQUESTED** [review]
waiting for a test case

**Comment by jjsjann123** [comment]
Test case added to test_cuda.py.
Verified test failure without the fix.

**Comment by jjsjann123** [comment]
FYI, We should close #22526 after merging this one.

**Review by ngimel — APPROVED** [review]
There's still lint.

**Comment by jjsjann123** [comment]
Lint complains about something in the generated file: 
```$ flake8
./test/test_quantization.py:4:1: F401 'torch.quantization as tq' imported but unused
The command "flake8" exited with 1.```

Maybe it's caused by the added test in common_nn.py? I don't see anything wrong there tho and lint doesn't give enough hint :/

**Comment by ngimel** [comment]
Yeah, it's probably unrelated then.

**Comment by jjsjann123** [comment]
OK, flake8 seems to be very specific about one line that is >120, while it's OK with other lines in the same manner in that function. There seems to be something funky there.

I can change that line just to bypass the pylinter. But I don't think it's relevant to this PR and probably would conflict with other PRs doing the same thing. Maybe we should just ignore it here?

**Comment by ngimel** [comment]
linter is broken by recently merged #21860, it has a few lines that are longer than 120 characters.

**Comment by ngimel** [comment]
lint should now be fixed, all other tests are green.

**Pull request closed without merging**